Makefile

```diff
@@ -148,9 +148,20 @@ else
 		$(args)
 endif
 
+unit_tests_api_keys: ## run unit tests only with api key tests
```

Comment:
I don't think we should classify any tests that require API keys as unit tests. 

And, if we don't, then we also go with my above suggestion to, by default, assume `integration_tests` uses api keys, and rename the below to:

```
integration_tests: ## Run all integration tests
integration_tests_api_key_required_only: ## Run only integration tests that require api keys
integration_tests_no_api_key_required_only: ## .. 
```